src/libsyntax/expand/mod.rs

```diff
@@ -9,8 +9,6 @@ bitflags::bitflags! {
     /// Built-in derives that need some extra tracking beyond the usual macro functionality.
     #[derive(Default)]
     pub struct SpecialDerives: u8 {
-        const PARTIAL_EQ = 1 << 0;
-        const EQ         = 1 << 1;
         const COPY       = 1 << 2;
```

Comment:
Is there a reason this can't be `1 << 0` now? (Or `SpecialDerives` eliminated entirely, if `COPY` is the only one.)

Reply:
I don't think the value particularly matters. When I was writing this I figured that it was best to leave the value alone, to keep the diff minimal. But since it seems like changing it to `1 << 0` should solely affect this line and nothing else, it seems like it would do no harm to make that change.

Getting rid of `SpecialDerives` entirely is a slightly different matter. If you mean "replace the bitflag with a single boolean": Sure, we *could* do that, but I'm not sure it adds much value.

If you mean get rid of this state entirely, that sounds nicer indeed. But I do not know enough about what the `Copy` flag is doing here to know how hard that might be.

Reply:
> Getting rid of `SpecialDerives` entirely is a slightly different matter. If you mean "replace the bitflag with a single boolean": Sure, we _could_ do that, but I'm not sure it adds much value.

I think it does; it gets rid of a type from libsyntax that doesn't *really* belong there. A `bool` would be good.

> But I do not know enough about what the `Copy` flag is doing here to know how hard that might be.

If I recall correctly, this exists so that `derive(Clone, Copy)` can be optimized to use `Copy` for `Clone`.

Reply:
FWIW, I'll cleanup this to `bool` (or rather `FxHashSet`) this week if it's not done in this PR.

Reply:
Okay, I'm convinced that getting rid of `SpecialDerives` has value, if only to stop others from extending it with more special derives in the future.

Let me see what I can do this morning.